Content.Server/GameObjects/EntitySystems/CuffableSystem.cs

```diff
@@ -0,0 +1,51 @@
+﻿#nullable enable
+using Content.Server.GameObjects.Components.ActionBlocking;
+using Content.Server.GameObjects.Components.GUI;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+
+namespace Content.Server.GameObjects.EntitySystems
+{
+    [UsedImplicitly]
+    public class CuffableSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            EntityManager.EventBus.SubscribeEvent<HandCountChangedEvent>(EventSource.Local, this, OnHandCountChanged);
+        }
+
+        /// <summary>
+        ///     Check the current amount of hands the owner has, and if there's less hands than active cuffs we remove some cuffs.
+        /// </summary>
+        private void OnHandCountChanged(HandCountChangedEvent message)
+        {
+            var owner = message.Sender;
+
+            if (!owner.TryGetComponent(out CuffableComponent? cuffable) ||
+                !cuffable.Initialized) return;
+
+            var dirty = false;
+            var handCount = owner.GetComponentOrNull<HandsComponent>()?.Count ?? 0;
+
+            while (cuffable.CuffedHandCount > handCount && cuffable.CuffedHandCount > 0)
+            {
+                dirty = true;
+
+                var container = cuffable.Container;
+                var entity = container.ContainedEntities[^1];
+
+                container.Remove(entity);
+                entity.Transform.WorldPosition = owner.Transform.Coordinates.Position;
```

Comment:
This won't work if the the owner is parented to something else like a locker.

Reply:
Fixed in https://github.com/space-wizards/space-station-14/pull/3321/commits/b0e00122c7092f5bb993b4a084b605d222dd94bc